llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -3289,7 +3285,7 @@ LegalizerHelper::widenScalar(MachineInstr &MI, unsigned TypeIdx, LLT WideTy) {
       Observer.changingInstr(MI);
 
       widenScalarSrc(
-          MI, LLT::vector(VecTy.getElementCount(), WideTy.getSizeInBits()), 1,
```

Comment:
The cases directly using LLT::vector/fixed_vector are slightly different, since this doesn't handle scalar inputs which are presumed impossible. This is a direct replacement for the scalarOrVector cases